fuel-tx/src/transaction/types/input/coin.rs

```diff
@@ -30,6 +37,19 @@ mod private {
 }
 
 /// Specifies the coin based on the usage context. See [`Coin`].
+#[cfg(feature = "da-compression")]
+pub trait CoinSpecification: private::Seal {
+    type Witness: AsField<u16>
+        + for<'a> Compressible<Compressed: Clone + Serialize + Deserialize<'a>>;
+    // TODO: We need to compress the predicate as well
```

Comment:
We also need to compress predicates, it is another good field for optimization besides script

Reply:
This will be a breaking change, since we have to add a wrapper type for the predicate, and the field is public. (If we were using the attribute-based approach, this wouldn't be an issue.)